### ISSUE TITLE
add support for wayland on linux

### ISSUE THREAD

**Issue by SmitTheTux** [brave/brave-browser#6212]

**Comment by kylew01** [comment]
I second that! I love using Brave on Linux, but with fractional scaling everything is blurry.

**Comment by fmarier** [comment]
There is work upstream to make Wayland on Linux a fully-supported platform: https://groups.google.com/a/chromium.org/d/topic/chromium-dev/-B8JhTojOtM/discussion

**Comment by pswu11** [comment]
I also just started to Brave it's amazing. But without wayland support it looks very blurry. It would be perfect to have web3.0 browser with native wayland support. Thank you!

**Comment by Isselbaecher** [comment]
I support this feature request! :)

**Comment by fuzunspm** [comment]
Please support wayland!

**Comment by tsujp** [comment]
+1 not supporting wayland means I am going back to Chrome.

**Comment by yavorski** [comment]
@tsujp I don't think chrome or chromium supports wayland.
The issue in the chromium repo is still open https://bugs.chromium.org/p/chromium/issues/detail?id=578890
Anyway you can go back and forward to whatever you want :)

**Comment by tsujp** [comment]
@yavorski I stand corrected, you are right the Chromium build from `community` is indeed using XWayland. My mistake there.

Might play around with that and Brave then :)

**Comment by SmitTheTux** [comment]
@tsujp if u are using arch or manjaro, try chromium-ozone from AUR or prebuilt binary from chaotic-aur repo. Its quite usable

**Comment by tsujp** [comment]
@SmitTheTux solid recommendation `chromium-ozone` is running even better so smooth.

**Comment by rohitghali** [comment]
+1 for this. I use full wayland, and I love brave. I want to use it in wayland too.

**Comment by Edward-0** [comment]
+1 for this. I use full wayland, and I love brave. I want to use it in wayland too.

**Comment by DrSh4dow** [comment]
+1 this is most likely a needed feature

**Comment by Ella-0** [comment]
Yeah especially as upstream chromium is getting more and more support for it.

**Comment by fourstepper** [comment]
Did anyone try the dev/nightlies for the Wayland support?

**Comment by kcalvelli** [comment]
> Did anyone try the dev/nightlies for the Wayland support?

I've been running Nightly (1.18.37) in a Wayland session on Plasma 5.20 today and seems to be working okay so far.  Using Exec=/usr/bin/brave-browser-nightly --enable-features=UseOzonePlatform --ozone-platform=wayland in my desktop file to launch.  PWAs are a little weird, as they get a system title bar and don't use the GTK theme (so get two sets of close buttons).  Brave Together works, but cannot share screen or app window (only browser tab).

**Comment by Ella-0** [comment]
I've been trying out the nightly too and it seams to work really well. As for the screen recording issues @kcalvelli mentioned. I don't think this is a Brave issue and more likely that pipewire hasn't been setup in the correct way which is very easy to do. It took me ages to get pipewire working properly with sway. I'm able to get all pipewire application windows and the desktop using `xdg-desktop-portal-wlr`

**Comment by fourstepper** [comment]
I am on GNOME, Wayland seems to work on the new release but the window each window is not being propagated to the panel, so I cannot switch apps with my keyboard shortcuts

**Comment by NullSense** [comment]
@Ella-0 I have pipewire set up to work on chromium and firefox, but it doesn't seem to work with the latest nightly and wayland enabled.

**Comment by fourstepper** [comment]
@NullSense what is the indication that pipewire doesn't work for you? What version are you on?

**Comment by NullSense** [comment]
@fourstepper 

```
❯ pacman -Qs pipewire
local/libpipewire02 0.2.7-1
local/pipewire 0.3.15-2
```

**Comment by fourstepper** [comment]
Are you talking about screen sharing of the whole desktop, window or tabs?

**Comment by NullSense** [comment]
> Are you talking about screen sharing of the whole desktop, window or tabs?

The whole desktop, windows **don't** work, only sharing tabs.

**Comment by fourstepper** [comment]
I can reproduce that as well

**Comment by NullSense** [comment]
> I can reproduce that as well

https://wiki.archlinux.org/index.php/PipeWire#WebRTC_screen_sharing According to the Arch wiki, it should be easy to enable like this, but I never got it working.

**Comment by fourstepper** [comment]
Didn't read the Arch wiki, but enabling the flag worked for me on GNOME in Jitsi Meet

![image](https://user-images.githubusercontent.com/49439044/98926816-12178800-24d8-11eb-9b24-18cbf3db5841.png)

**Comment by NullSense** [comment]
> Didn't read the Arch wiki, but enabling the flag worked for me on GNOME in Jitsi Meet
> 
> ![image](https://user-images.githubusercontent.com/49439044/98926816-12178800-24d8-11eb-9b24-18cbf3db5841.png)

Strange, I got it to work if I run these commands:

```
❯ systemctl --user stop pipewire.service pipewire.socket xdg-desktop-portal.service xdg-desktop-portal-wlr.service
env PIPEWIRE_DEBUG=3 pipewire
/usr/libexec/xdg-desktop-portal-wlr -l DEBUG
/usr/libexec/xdg-desktop-portal --verbose -r
```

**Comment by NullSense** [comment]
As another finding: I noticed speed reader does not work on my machine (wayland mode as well), can anyone reproduce?

**Comment by yavorski** [comment]
Chromium version 87 supports native Wayland with the following flags:

```ini
--enable-features=UseOzonePlatform --ozone-platform=wayland
```

**Comment by why-not-try-calmer** [comment]
It's not very difficult to launch Brave on on a full wayland session. I am using KDE Plasma 5.21:
```
$  GDK_BACKEND=wayland (or fish: set GDK_BACKEND x11)
$  brave-browser
```
Of course it would be much nicer to have native support.

**Comment by sith-on-mars** [comment]
On my OpenSUSE Tumbleweed, the screen sharing doesn't work. I clicked the button, but nothing happened. I have enabled WebRTC screen sharing and launched Brave with the following command: `brave-browser --enable-features=UseOzonePlatform --ozone-platform=wayland`. The screen sharing works on Chromium however.

**Comment by Ella-0** [comment]
you might need to enable WebRTC and Pipewire support in brave://flags.

**Comment by Ella-0** [comment]
Is chromium running with Wayland?

**Comment by sith-on-mars** [comment]
> Is chromium running with Wayland?

Yes I use Wayland on my OpenSUSE Tumbleweed. I launched Brave with the command `brave-browser --enable-features=UseOzonePlatform --ozone-platform=wayland`.

**Comment by WhyNotHugo** [comment]
> Is chromium running with Wayland?

It "runs" as of a few stable releases, however, many things are broken.  
As of v89 (still no stable release for it), screen sharing works. AFAIK, that's the last feature that's not working on Wayland but working on other platforms.

**Comment by Borges-Filipe** [comment]
I launched Brave with:

`brave --enable-features=UseOzonePlatform --ozone-platform=wayland`

The mouse cursor becomes tiny but everything else looks crisp even with fractional scaling turned on. 
I am on Manjaro Gnome using Wayland.

Any way to fix the cursor?

**Comment by microo8** [comment]
It works, you can add these flags in `.config/brave-flags.conf`.
No Xwayland is running :) but some sites don't work. eg playing a youtube video :(

**Comment by maarteNNNN** [comment]
I did `brave://flags`, searched for `wayland` and enabled the `WebRTC PipeWire support` and it worked.
![image](https://user-images.githubusercontent.com/14275291/123105698-6095e900-d40e-11eb-8609-9033b2b9f656.png)

**Comment by mastertinner** [comment]
> I launched Brave with:
> 
> `brave --enable-features=UseOzonePlatform --ozone-platform=wayland`
> 
> The mouse cursor becomes tiny but everything else looks crisp even with fractional scaling turned on.
> I am on Manjaro Gnome using Wayland.
> 
> Any way to fix the cursor?

Interesting... For me, the cursor gets huge... But I have scaling set to 2.0.

**Comment by WhyNotHugo** [comment]
The way fractional scaling works is by rendering larger and then downscaling. E.g.: a 1.5x scaling actually scales to 3x and then downscales that by 50%.

I can't image why 2x scaling would render it larger, but fractional scaling renders it smaller. Unless you're using fractional scaling < 1?

**Comment by ahydronous** [comment]
@microo8 
> It works, you can add these flags in `.config/brave-flags.conf`.
> No Xwayland is running :) but some sites don't work. eg playing a youtube video :(

Don't say random stuff without testing. I just tried this on Fedora 34 and it did absolutely nothing.

**Comment by mielouk** [comment]
This is a key feature for a browser to be used on any distro used in the foreseeable future. 

Please support Wayland :crystal_ball:

**Comment by Schmensch** [comment]
Pleeease add this. I think upstream chromium is working on a simple toggle in chrome://flags to toggle Wayland via Ozone.

**Comment by Pound-Hash** [comment]
> I did `brave://flags`, searched for `wayland` and enabled the `WebRTC PipeWire support` and it worked using Ubuntu 21.04. ![image](https://user-images.githubusercontent.com/14275291/123105698-6095e900-d40e-11eb-8609-9033b2b9f656.png)

What's this for? I tried it on Arch, and it didn't help loading the browser.

**Comment by Schmensch** [comment]
@Pound-Hash that's not for running Brave Wayland-Native, you can do that with `brave --enable-features=UseOzonePlatform --ozone-platform=wayland`. 

WebRTC PipeWire support is required for screen-sharing on Wayland. It's independent from Ozone.

**Comment by hugosxm** [comment]
Hi, juste to let you know but on sway, the claim captcha is not working, i have to log off, log on to xfce, claim my bat then come back to sway. Not a big problem but worth noting. Also i would be happy to not use xwayland anymore ^^

**Comment by fmarier** [comment]
@hugosxm also see https://github.com/brave/brave-browser/issues/20428 for the CAPTCHA issues on Linux

**Comment by enrico223** [comment]
> I did `brave://flags`, searched for `wayland` and enabled the `WebRTC PipeWire support` and it worked using Ubuntu 21.04. ![image](https://user-images.githubusercontent.com/14275291/123105698-6095e900-d40e-11eb-8609-9033b2b9f656.png)

I'm on fedora 35, you can also choose Wayland from the "preferred Ozone platform" just to be sure. It works.

**Comment by DaWe35** [comment]
> I did `brave://flags`, searched for `wayland` and enabled the `WebRTC PipeWire support` and it worked using Ubuntu 21.04. ![image](https://user-images.githubusercontent.com/14275291/123105698-6095e900-d40e-11eb-8609-9033b2b9f656.png)

OMG this worked, why Brave is unable to detect Wayland automatically?

**Comment by why-not-try-calmer** [comment]
> > I did `brave://flags`, searched for `wayland` and enabled the `WebRTC PipeWire support` and it worked using Ubuntu 21.04. ![image](https://user-images.githubusercontent.com/14275291/123105698-6095e900-d40e-11eb-8609-9033b2b9f656.png)
> 
> OMG this worked, why Brave is unable to detect Wayland automatically?

It's the same as with all other Chromium-based browsers.

On that, I think that this thread has outlived its purpose. Brave does fully support Wayland on Linux using the right flags.

**Comment by All3xJ** [comment]
You can just set the flag "ozone-platform-hint" to Wayland, in brave://flags. 

![image](https://github.com/brave/brave-browser/assets/20903913/eecb8968-a06a-4e5b-9191-e5292a8fa342)

**Comment by gantryyork** [comment]
I don't know if this is the issue, but Brave can't be resized with the mouse using the Cosmic (Rust) desktop.

**Comment by izmyname** [comment]
> You can just set the flag "ozone-platform-hint" to Wayland, in brave://flags.
> 
> ![image](https://private-user-images.githubusercontent.com/20903913/252147950-eecb8968-a06a-4e5b-9191-e5292a8fa342.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIzMDEyMzEsIm5iZiI6MTcxMjMwMDkzMSwicGF0aCI6Ii8yMDkwMzkxMy8yNTIxNDc5NTAtZWVjYjg5NjgtYTA2YS00ZTViLTkxOTEtZTUyOTJhOGZhMzQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA1VDA3MDg1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2ZWMwMjg1ZWFjNjBhYzc3MDgxZjkwZWQ3ODFjNTU5YTBjYTYxZThiNmU3NGU0MTIzOGY0NjlmMTZlNjlkMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oHK5olNe-63kAVtOXh9yrZhfi1kYfFp_wgfOlIg8ktU)

Doesn't work in flatpak version.

**Comment by RedBlizard** [comment]
Latest recent update on April 17 2024  version 1.65.114  is missing something regarding wayland support now you have to set this again in the brave-browser.desktop  to make sure  brave launches correctly Exec=brave --enable-features=UseOzonePlatform --ozone-platform=wayland  with the previous version this was not needed so looks like some wayland support is missing , lucky the flag is still there. Dont why this has changed.

**Comment by Mikaela** [comment]
`--ozone-platform-hint` not working was https://github.com/brave/brave-browser/issues/37498

**Comment by DaWe35** [comment]
@RedBlizard you're a legend, thank you. Brave just glitched after an update, but `brave --enable-features=UseOzonePlatform --ozone-platform=wayland` works well 🙏🙏🙏

**Comment by RedBlizard** [comment]
> @RedBlizard you're a legend, thank you. Brave just glitched after an update, but `brave --enable-features=UseOzonePlatform --ozone-platform=wayland` works well 🙏🙏🙏

What you also can do is create in .config this flag file brave-flags.conf and only add this line !   --enable-features=UseOzonePlatform --ozone-platform=wayland  that is the cleaner solution.

**Comment by parasnailor** [comment]
Brave works great under Wayland after configuring the feature flag, but I am curious as to why it doesn't autodetect Wayland. It's defaulting to XWayland on Kubuntu 25.04 for me.

**Comment by jassielof** [comment]
This issue should be solved as of latest Brave/Chromium, since it has finally defaulted to Wayland.
https://www.reddit.com/r/linux/comments/1moyx7f/chromium_141_will_now_use_wayland/

**Comment by Nyunyun** [comment]
I had to revert it by adding `--ozone-platform=x11`  to `~/.config/brave-flags.conf`:

With me on Arch, latest KDE and an RTX 4090 with 590 driver there are some rendering issues, e.g. [this webm](https://s1.pearlcdn.com/global_static/video/cd_pre_order_main_loop.webm) appears in green with a few artifacts.

The last time I tested Wayland Brave (June 2025), there were also issues with randomly not loading the complete webpage. I didn't test long enough now to confirm if this still happens, but I guess it is still worth mentioning.

**Comment by sidalsolgun** [comment]
The year is 2026, and the self-identifying "secure" browser still uses X11 ?!

https://flathub.org/en/apps/com.brave.Browser
